src/encoded/commands/generate_ontology.py

```diff
@@ -441,38 +449,69 @@ def connect2server(env=None, key=None):
+        if 'term_name' in term and term['term_name'].lower().startswith('obsolete'):
+            continue
+        live_terms[termid] = term
+    return live_terms
 
 
-def verify_and_update_ontology(terms, ontologies, dbterms):
-    '''checks to be sure the ontology associated with the term agrees with
-        the term prefix as long as term does not already exist in db.
-        If it doesn't it is likely that the term was
-        imported into a previously processed ontology and so the ontlogy
-        of the term should be updated to the one that matches the prefix
-    '''
-    ont_lookup = {o['uuid']: o['ontology_prefix'] for o in ontologies}
-    ont_prefi = {v: k for k, v in ont_lookup.items()}
-    to_delete = []
+def _format_def_str(defdict):
+    dstring = ''
+    for d, o in sorted(defdict.items()):
+        ostr = ', '.join(sorted([ostr.strip() for ostr in o]))
+        dstring += '{} ({}) '.format(d, ostr)
+    return dstring.rstrip()
+
+
+def set_definition(terms, ontologies):
+    uuid2prefix = {o.get('uuid'): o.get('ontology_prefix') for o in ontologies}
     for termid, term in terms.items():
-        # if termid in dbterms:
-        #     if ont_lookup.get(term['source_ontology'], None) != dbterms[termid]['source_ontology']:
-        #         to_delete.append(termid)
-        if ont_lookup.get(term['source_ontology'], None):
-            prefix = termid.split(':')[0]
-            if prefix in ont_prefi:
-                if prefix != ont_lookup[term['source_ontology']]:
-                    term['source_ontology'] = ont_prefi[prefix]
-        if termid in dbterms and term['source_ontology'] != dbterms[termid]['source_ontology']['uuid']:
-            to_delete.append(termid)
-            print('WARNING - {} is already present as a term in {} and not {}'.format(termid, dbterms[termid]['source_ontology']['display_title'], term['source_ontology']))
-    keep_terms = {k: v for k, v in terms.items() if k not in to_delete}
-    return keep_terms
+        definition = {}
+        found = False
+        tdefs = term.get('definitions')
+        if not tdefs:
+            continue
+        fromont = [p for p in tdefs.keys() if termid.startswith(p)]
+        if fromont:
+            ont = fromont[0]
+            definition[ont] = tdefs[ont]
+        elif len(tdefs) == 1:
+            definition = tdefs  # we've got a single value to deal with
+        for p, defs in definition.items():
+            # ideal case where term is directly from the ontology based on termid prefix
+            term['definition'] = '{} ({})'.format(' -- '.join(sorted(list(set([d.strip() for d in defs])))), p)
+            found = True
+        if found:
+            del term['definitions']
+            continue
+        # if we get here add all the defs by uniquifying with onts
+        for p, defs in tdefs.items():
+            for d in defs:
+                definition.setdefault(d.strip(), []).append(p)
+        dstring = _format_def_str(defintion)
```

Comment:
looks like a typo (`defintion` should be `definition`)

Reply:
fixed.